### PR TITLE
[Bug] Fix dead lock in olap scan node and refactor some code in FE profile

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

RT

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [x ] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yangzhg — DISMISSED** [review]
LGTM

File: be/src/exec/olap_scan_node.cpp [hunk 1]
```diff
@@ -1470,12 +1470,12 @@ void OlapScanNode::scanner_thread(OlapScanner* scanner) {
 
     _scan_cpu_timer->update(cpu_watch.elapsed_time());
     _scanner_wait_worker_timer->update(wait_time);
-    _scan_batch_added_cv.notify_one();
 
     // The transfer thead will wait for `_running_thread==0`, to make sure all scanner threads won't access class members.
     // Do not access class members after this code.
     std::unique_lock<std::mutex> l(_scan_batches_lock);
     _running_thread--;
+    _scan_batch_added_cv.notify_one();
```
Comment (morningman) [inline be/src/exec/olap_scan_node.cpp:1478]: Add some comment to explain?

Comment (HappenLee) [reply be/src/exec/olap_scan_node.cpp:1478]: OK

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**